pydis_core/utils/paste_service.py

```diff
@@ -0,0 +1,111 @@
+from typing import TypedDict
+
+from aiohttp import ClientConnectorError, ClientSession
+
+from pydis_core.utils import logging
+
+log = logging.get_logger(__name__)
+
+FAILED_REQUEST_ATTEMPTS = 3
+MAX_PASTE_SIZE = 128 * 1024  # 128kB
+
+
+class PasteResponse(TypedDict):
+    """
+    A successful response from the paste service.
+
+    args:
+        link: The URL to the saved paste.
+        removal: The URL to delete the saved paste.
+    """
+
+    link: str
+    removal: str
+
+
+class PasteUploadError(Exception):
+    """Raised when an error is encountered uploading to the paste service."""
+
+
+class PasteTooLongError(Exception):
+    """Raised when content is too large to upload to the paste service."""
+
+
+async def send_to_paste_service(
+    *,
+    contents: str,
+    paste_url: str,
+    http_session: ClientSession,
+    file_name: str = "",
+    lexer: str = "python",
+    max_size: int = MAX_PASTE_SIZE,
+) -> PasteResponse:
+    """
+    Upload some contents to the paste service.
+
+    Args:
+        contents: The content to upload to the paste service.
+        paste_url: The base url to the paste service.
+        http_session (aiohttp.ClientSession): The session to use when POSTing the content to the paste service.
+        file_name: The name of the file to save to the paste service.
+        lexer: The lexer to save the content with.
+        max_size: The maximum number of bytes to be allowed. Anything larger that 256kB will be rejected.
```

Comment:
This needs to be updated.
```suggestion
        max_size: The maximum number of bytes to be allowed. Anything larger that 128kB will be rejected.
```

Reply:
Updated in the [forcepush](https://github.com/python-discord/bot-core/compare/9478a6fcbda210b7b8ad4ee83bc8ba3e89ff5498..7345b10961af47ca179d8b7009c0d7d0eb3c08e2) to reference the constant instead. Looks like this now

![image](https://github.com/python-discord/bot-core/assets/9753350/cbb89b17-dd1c-4742-a6cb-3fa514a8d893)

![image](https://github.com/python-discord/bot-core/assets/9753350/86a28368-512b-4519-9d4b-ede01450721b)